src/compiler/checker.ts

```diff
@@ -32050,11 +32097,29 @@ namespace ts {
             if (leftType === silentNeverType || rightType === silentNeverType) {
                 return silentNeverType;
             }
-            leftType = checkNonNullType(leftType, left);
+            if (isPrivateIdentifier(left)) {
+                if (languageVersion < ScriptTarget.ESNext) {
+                    checkExternalEmitHelpers(left, ExternalEmitHelpers.ClassPrivateFieldIn);
+                }
+                // Unlike in 'checkPrivateIdentifierExpression' we now have access to the RHS type
+                // which provides us with the opportunity to emit more detailed errors
+                if (!getNodeLinks(left).resolvedSymbol && getContainingClass(left)) {

```

Comment:
why is it important to check `getContainingClass(left)`?

Reply:
In this case:

```ts
class X {
  #field;
}

function f(v: X) {
  return #field in v;
}
```

Without this additional check we get two errors:

- Property '#field' does not exist on type 'X'.
- Private identifiers are not allowed outside class bodies

It felt better to ensure the focus was only on the second one. Though happy to drop it and save some CPU.